### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by X-Elie-X**
### In this PR I implemented the following:
- Create a class Classroom
- Create the has-many/belongs-to relationship between Classroom and the Student
- Create a class Book with the following
- Create a class Rental with the following
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class Book
+  attr_accessor :title, :author, rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_book(person, date)
```
Comment (tufoinnkuo10) [inline book.rb]: - [ ] nice work. This method should be for adding rental and not book. please check that.

File: person.rb [hunk 2]
```diff
@@ -22,4 +24,8 @@ def to_age?
   def correct_name
     @name
   end
+
+  def add_book(person, date)
```
Comment (tufoinnkuo10) [inline person.rb]: - [ ] same here. the method should be `add_rental` instead of `add_book`

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @X-Elie-X 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by Ugboaja-Uchechi — APPROVED** [review]
Hi @X-Elie-X, 

## Status: APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 
- Great job with your design and functionalities 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**